src/commands/prompt/index.ts

```diff
@@ -207,20 +180,23 @@ async function runInternal(initialPrompt: string, options: PromptOptions) {
   // eslint-disable-next-line no-constant-condition
   while (true) {
     const userPrompt = await inputLine('me: ');
-    const isCommand = processCommand(userPrompt, { messages, providerName: provider.name, config });
+    const isCommand = processCommand(session, userPrompt);
     if (isCommand) {
       continue;
     }
 
     output.outputAiProgress('Thinking...');
 
-    messages.push({ role: 'user', content: userPrompt });
-    const { messageText, stats, response } = await provider.getChatCompletion(config, messages);
+    session.messages.push({ role: 'user', content: userPrompt });
```

Comment:
Processing user interaction inside the loop and outside it seems identical. Maybe we could extract it to a separate "processUserMessage" function or something like that?